clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp

```diff
@@ -0,0 +1,54 @@
+//===--------------- QueryAPIMapping.cpp ----------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "QueryAPIMapping.h"
+
+namespace clang {
+namespace dpct {
+
+std::unordered_map<std::string, std::string> APIMappingEntrys::EntryMap;
+
+void APIMappingEntrys::registerEntry(const std::string &Name,
+                                     const std::string &Description) {
+  EntryMap[Name] = Description;
+}
+
+void APIMappingEntrys::initEntryMap() {
+  static const std::string MultiStr =
+      "\nThere are multi migration solutions for this API with different "
+      "migration options,\nsuggest to use the tool to migrate a complete test "
+      "case to see more detail of the migration.";
+#define ENTRY(INTERFACENAME, APINAME, VALUE, FLAG, TARGET, COMMENT, MAPPING)   \
+  registerEntry(#APINAME, MAPPING);
+#define ENTRY_MEMBER_FUNCTION(OBJNAME, INTERFACENAME, APINAME, VALUE, FLAG,    \
+                              TARGET, COMMENT, MAPPING)                        \
```

Comment:
and if mapping if "NA", no need to build into the map.

Reply:
fixed